test/cypress/integration/worldpay-3ds-flex-ddc/payment-with-session-id.spec.js

```diff
@@ -95,32 +95,36 @@ describe('Worldpay 3ds flex card payment flow', () => {
   }
 
   setUpAndCheckCardPaymentPage()
-  describe('Secure confirmation page', () => {
-    it('Submitting confirmation with valid details and a JWT should cause worldpay iFrame result to post a message with the session ID', () => {
+  describe('Valid DDC response', () => {
+    it('Form is submitted with the session ID from the DDC response', () => {
       cy.task('setupStubs', [...confirmPaymentDetailsStubs, worldpay3dsFlexDdcStub])
 
       cy.get('#card-details').submit().should($form => {
         const formVal = $form.first()[0].elements.worldpay3dsFlexDdcResult.value
         expect(formVal).to.eq(worldpaySessionId)
+        const ddcStatusVal = $form.first()[0].elements.worldpay3dsFlexDdcStatus.value
+        expect(ddcStatusVal).to.eq('valid DDC result')
       })
     })
   })
 
   setUpAndCheckCardPaymentPage()
-  describe('Secure confirmation page', () => {
-    it('If Worldpay were to respond with status=false, submitting confirmation should not include the hidden input containing the session ID', () => {
+  describe('Worldpay responds with status=false', () => {
+    it('submitting confirmation should not include the hidden input containing the session ID', () => {
       cy.task('setupStubs', [...confirmPaymentDetailsStubs, worldpay3dsFlexDdcStubFailure])
 
       cy.get('#card-details').submit().should($form => {
         const formVal = $form.first()[0].elements.worldpay3dsFlexDdcResult
         expect(formVal).to.eq(undefined)
+        const ddcStatusVal = $form.first()[0].elements.worldpay3dsFlexDdcStatus.value
+        expect(ddcStatusVal).to.eq('DDC response had Status of false')
```

Comment:
Nice that you added these assertions but I’m quite tempted to ruin your day by asking you to add in scenarios for the other statuses we might report.

Reply:
My opinion is that would be overkill, we don't generally test logging but as these are set by client-side javascript I wanted to ensure that it didn't break in some way we didn't notice by someone changing something that links the components together.

Also can't do an assertion for the timeout in cypress tests, as it will just look at the value at the time the form is submitted.